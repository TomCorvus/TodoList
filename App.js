import React from 'react';
import { Provider } from 'react-redux';
import { StyleSheet, View, StatusBar, Platform } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import { AppNavigator } from './components/Navigation/AppNavigator';
import { store } from './reducers/TodoReducer';
import globalColors from './constants/Colors';

class App extends React.Component {
	constructor(props) {
		super(props);
	}

	componentDidMount() {}

	render() {
		return (
			<Provider store={store}>
				{Platform.OS === 'ios' && (
					<>
						<View style={styles.appBar}></View>
						<StatusBar barStyle="light-content" />
					</>
				)}
				<View style={styles.container}>
					<AppNavigator />
				</View>
			</Provider>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: globalColors.bodyLightBackgroundColor,
	},
	appBar: {
		height: getStatusBarHeight(),
		backgroundColor: globalColors.headerBackgroundColor,
	},
});

export default App;
