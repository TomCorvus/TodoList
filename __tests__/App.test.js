import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.useFakeTimers();

test('renders correctly', async () => {
	const tree = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});
