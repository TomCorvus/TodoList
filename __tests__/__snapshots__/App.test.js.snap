// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#012E40",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      onInsetsChange={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 0.01,
                    "restSpeedThreshold": 0.01,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#012E40",
                                "elevation": 4,
                                "flexDirection": "row",
                                "height": 60,
                                "paddingBottom": 10,
                                "paddingTop": 10,
                                "position": "relative",
                                "shadowColor": "#000",
                                "shadowOffset": Object {
                                  "height": 2,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.23,
                                "shadowRadius": 2.62,
                                "zIndex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                  "flexDirection": "row",
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignContent": "stretch",
                                    "alignItems": "center",
                                    "flex": 1,
                                    "justifyContent": "center",
                                    "paddingRight": 20,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  Chlourse
                                </Text>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "alignContent": "center",
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                  "width": 60,
                                  "zIndex": 2,
                                }
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "height": 60,
                                    "justifyContent": "center",
                                    "width": 60,
                                  }
                                }
                              >
                                <Text />
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "backgroundColor": "#F5F5F5",
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                  "padding": 0,
                                }
                              }
                            >
                              <RCTScrollView
                                closeOnRowBeginSwipe={false}
                                closeOnRowOpen={true}
                                closeOnRowPress={true}
                                closeOnScroll={true}
                                data={Array []}
                                directionalDistanceChangeThreshold={2}
                                disableHiddenLayoutCalculation={false}
                                disableLeftSwipe={false}
                                disableRightSwipe={true}
                                disableVirtualization={false}
                                getItem={[Function]}
                                getItemCount={[Function]}
                                horizontal={false}
                                initialNumToRender={10}
                                keyExtractor={[Function]}
                                leftOpenValue={0}
                                maxToRenderPerBatch={10}
                                numColumns={1}
                                onContentSizeChange={[Function]}
                                onEndReachedThreshold={2}
                                onLayout={[Function]}
                                onMomentumScrollEnd={[Function]}
                                onRowDidOpen={[Function]}
                                onScroll={[Function]}
                                onScrollBeginDrag={[Function]}
                                onScrollEndDrag={[Function]}
                                previewDuration={400}
                                previewFirstRow={false}
                                previewOpenDelay={600}
                                previewOpenValue={-50}
                                previewRepeat={false}
                                previewRepeatDelay={1000}
                                previewRowKey="0"
                                recalculateHiddenLayout={false}
                                refreshing={true}
                                removeClippedSubviews={false}
                                renderHiddenItem={[Function]}
                                renderItem={[Function]}
                                rightOpenValue={-150}
                                scrollEventThrottle={50}
                                stickyHeaderIndices={Array []}
                                swipeToClosePercent={50}
                                swipeToOpenPercent={50}
                                swipeToOpenVelocityContribution={0}
                                updateCellsBatchingPeriod={50}
                                useAnimatedList={false}
                                useNativeDriver={true}
                                viewabilityConfigCallbackPairs={Array []}
                                windowSize={21}
                              >
                                <View />
                              </RCTScrollView>
                            </View>
                            <View
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#4B8CA6",
                                  "flexDirection": "row",
                                  "height": 60,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "flexDirection": "row",
                                  }
                                }
                              >
                                <Text />
                                <Text
                                  style={
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  [missing "mock.addList" translation]
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        />
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;
